### DISCUSSION TITLE
System.Net.Mail.SmtpException: 'Failure sending mail.'

### DISCUSSION THREAD

**Discussion by luizgoebel** [balta-io/forum#165] [Tira Dúvidas] [answered]
Olá André, fiz toda a implementação conforme o curso
Fundamentos do ASP.NET 6,  porém estou tendo esse erro na hora de gerar account e enviar o email, pode me ajudar ?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @luizgoebel, acredito que o erro não veio (era um print?)

**Reply by luizgoebel** [reply]
![erro ao enviar email](https://user-images.githubusercontent.com/72111314/207603614-dbaeda20-c08e-40c0-b059-9eb5c84ac972.png)
agora sim, esta exceção

**Reply by andrebaltieri** [reply]
Bom dia, @luizgoebel, será que a porta está correta? Tem algum Firewall na sua máquina bloqueando? Pelo erro, é um problema de rede.

**Reply by luizgoebel** [reply]
estou testando com meu email (gmail), seguindo todos os passos do curso, testei outras formas porém sem sucesso

**Comment by luizgoebel** [comment]
Bom dia,
A função cria o usuário, cria o Mail Message mas na hora do send, retorna
essa exceção

Em qua., 14 de dez. de 2022 às 09:02, Andre Baltieri <
***@***.***> escreveu:

> Bom dia, @luizgoebel <https://github.com/luizgoebel>, acredito que o erro
> não veio (era um print?)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/balta-io/forum/discussions/165#discussioncomment-4399425>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARGFJUV4HGGONCM4YIY4NM3WNGZOZANCNFSM6AAAAAAS6MA64M>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Atenciosamente,*
*Luiz Eduardo Goebel - Desenvolvedor Web*
*Contato : (47) 99183-2512*


*"Se você desenvolve os hábitos do sucesso, você fará do sucesso um hábito"*

**Comment by luizgoebel** [comment]
Bom dia André,
revisei a porta, desativei o firewall porém ainda sem sucesso,
compreensível a minha dificuldade por que é uma skill que não tenho e quero
aprender mas preciso de ajuda mesmo.

[image: image.png]

Em qui., 15 de dez. de 2022 às 09:05, Andre Baltieri <
***@***.***> escreveu:

> Bom dia, @luizgoebel <https://github.com/luizgoebel>, será que a porta
> está correta? Tem algum Firewall na sua máquina bloqueando? Pelo erro, é um
> problema de rede.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/balta-io/forum/discussions/165#discussioncomment-4409494>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARGFJUUTWXUARZVUVLRTLVDWNMCOXANCNFSM6AAAAAAS6MA64M>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Atenciosamente,*
*Luiz Eduardo Goebel - Desenvolvedor Web*
*Contato : (47) 99183-2512*


*"Se você desenvolve os hábitos do sucesso, você fará do sucesso um hábito"*

**Comment by andrebaltieri** [comment]
Boa noite, @luizgoebel, como vai?
Só um dica, se puder responder pelo GitHub ao invés de responder por E-mail, fica melhor... suas imagens não estão aparecendo:
https://github.com/balta-io/forum/discussions/165

Além dos erros, você consegue colocar o trecho de código que está usando para enviar o E-mail aqui?

**Reply by luizgoebel** [reply]
Bom dia Balta, segue em anexo o trecho que esta ocorrendo o erro juntamente com o erro.
![erro ao enviar email](https://user-images.githubusercontent.com/72111314/208417951-47ca74e7-16f2-4f40-b036-c69aaa5cb865.png)

**Reply by andrebaltieri** [reply]
@luizgoebel muito obrigado!! Agora sim 😁
Este erro provavelmente ocorre por que o E-mail (From) é diferente do cadastrado no SendGrid e eles precisam ser iguais.
Na tier gratuita, para eviar Spam, eles só permitem envio de E-mails da conta padrão do sendgrid.

Pagando você pode verificar um domínio (Como temos com o @balta.io) e ai sem enviar usando um E-mail personalizado.

**Comment by dhionys-soares** [comment]
Olá Balta e Luiz, eu estava com o mesmo problema e consegui resolver. No sendgrid/configurações tem a autenticação de remetente, aí você vai e coloca o email que vai usar pra enviar o email, acho que de preferência, o cadastrado no sendgrid, depois de preencher, você confirma no seu email e testa novamente. Aqui comigo funcionou

**Reply by andrebaltieri** [reply]
Obrigado por compartilhar!
